src/evict.c

```diff
@@ -603,16 +603,19 @@ int performEvictions(void) {
                     if (current_db_keys == 0) continue;
 
                     total_keys += current_db_keys;
-                    do {
+                    for (l = 0; l < dbNonEmptySlots(db, keyType); l++) {
```

Comment:
let's cache the result of dbNonEmptySlots outside the loop.
(we can convert it to a `while(l--)` loop, or move to the initializer part of the `for` loop).